runtime/lua/vim/diagnostic.lua

```diff
@@ -1008,8 +1010,11 @@ M.handlers.virtual_text = {
 
       if virt_texts then
         api.nvim_buf_set_extmark(bufnr, virt_text_ns, line, 0, {
-          hl_mode = 'combine',
+          hl_mode = opts.virtual_text.hl_mode or 'combine',
           virt_text = virt_texts,
+          virt_text_pos = opts.virtual_text.virt_text_pos,
```

Comment:
Feels a bit redundant to have a property named `virt_text_pos` inside `virt_text`, but it does keep the consistent names with `nvim_buf_set_extmark`, which I think is better...